narwhals/_pandas_like/dataframe.py

```diff
@@ -268,15 +269,52 @@ def join(
         self,
         other: Self,
         *,
-        how: Literal["left", "inner", "outer"] = "inner",
-        left_on: str | list[str],
-        right_on: str | list[str],
+        how: Literal["left", "inner", "outer", "cross"] = "inner",
+        left_on: str | list[str] | None = None,
+        right_on: str | list[str] | None = None,
     ) -> Self:
         if isinstance(left_on, str):
             left_on = [left_on]
         if isinstance(right_on, str):
             right_on = [right_on]
 
+        if how == "cross":
+            if self._implementation in {"modin", "cudf"} or (
+                self._implementation == "pandas"
+                and (pd := get_pandas()) is not None
+                and parse_version(pd.__version__) < parse_version("1.2.0")
```

Comment:
is this right? the pandas docs say it's new in 1.2.0, but I guess it's a typo, the 1.3 docs don't have it https://pandas.pydata.org/pandas-docs/version/1.3/reference/api/pandas.DataFrame.join.html, and it only appears in 1.4